glean-core/ios/Glean/Net/HttpPingUploader.swift

```diff
@@ -108,6 +109,7 @@ public class HttpPingUploader {
                     glean_process_ping_upload_response(&incomingTask, result.toFfi())
                 }
             case .wait:
+                sleep(Constants.throttleBackoffMs)
```

Comment:
huh, I just looked into the code for iOS and we call `HttpPingUploader.process` synchronously in the task of the `initialize` function.
This means we essentially _block_ full initialization until we uploaded all pings. Now previously we were probably lucky in that if we're throttled we quickly get to 3 waits in a row and bail out, but now this could lead to blocking Glean initialization for an extended amount of time.

I think we need to look into that and ensure we're not doing that.
The intended behavior should be the same across platforms:
We launch a background worker to upload what's there. On Android we already do that.

Reply:
Okay, I opened a bug about this and will leave this PR blocked for now, until that is fixed.

Reply:
The bug: https://bugzilla.mozilla.org/show_bug.cgi?id=1668820